towerlib/entities/job.py

```diff
@@ -1628,6 +1758,53 @@ def stdout(self):
 
     # TOFIX use, model and implement scm_inventory_updates.
 
+    def monitor(self, update_id, timeout=None, interval=0.25):
+        """Monitors the project update status based on response id of the update call.
+
+        Returns:
+            string: Job status of the update
+
+        """
+        started = time.time()
+        next_line = 0
+
+        project_update = self._tower.get_project_update_by_id(update_id)
+        while True:
```

Comment:
This while loop would be much more elegant if it was a context manager since it is reused elsewhere.